src/AVFoundation/AVMetadataObject.cs

```diff
@@ -36,7 +36,7 @@ namespace AVFoundation {
 	public partial class AVMetadataObject {
 		public AVMetadataObjectType Type {
 			get {
-				return ObjectToEnum (WeakType);
+				return AVMetadataObjectTypeExtensions.GetValue (WeakType);
```

Comment:
Noob question - does the `AVMetadataObjectTypeExtension` get generated by the generator? I can't seem to find it in my local repo. Also, does it get generated for every `enum` with `[Field]` attributes defined?

Reply:
@prmallic yes, that's what makes it a _smart_ enum - when `[Field]` are added to enum members then the generator will create the conversion methods (NSString->enum and enum->NSString) automagically.